### PR TITLE
Update README and CONTRIBUTING formatting

### LINKED ISSUE #1034: [Bug]: Deprecated requirements
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We are mentioning `NodeJS version 14 and NPM` in the [README.md](https://github.com/opendatahub-io/odh-dashboard#requirements) even though with the latest changes added in #1033 we are now using `Node 18 and npm 8`.

We should update the versioning in order to reflect those changes.

### Expected Behavior

To have the correct versioning in our documentation.

### Steps To Reproduce

_No response_

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

_No response_

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #1034 

## Request review criteria:

- [X] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,74 +3,98 @@
...
-* [NodeJS and NPM](https://nodejs.org/)
-* [OpenShift CLI](https://docs.openshift.com/enterprise/3.2/cli_reference/get_started_cli.html#installing-the-cli)
-* [kustomize](https://github.com/kubernetes-sigs/kustomize)
-* [podman](https://github.com/containers/podman)
-* and have access to [Quay.io](https://quay.io/)
+
+- [NodeJS and NPM](https://nodejs.org/)
+  - Node recommended version -> `18.14.0`
+  - NPM recommended version -> `8.19.4`
+- [OpenShift CLI](https://docs.openshift.com/enterprise/3.2/cli_reference/get_started_cli.html#installing-the-cli)
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply CONTRIBUTING.md:12]: Yeah, you are right, completely missed this!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,74 +3,98 @@
...
 ## Writing code
+
 ### Running locally
+
 Development for both "frontend" and "backend" can be done while running
-```
-$ npm run dev
+
+``` bash
+npm run dev
```
Comment (andrewballantyne) [inline CONTRIBUTING.md:24]: This is good in a pinch -- but it sucks for actual development. You can't stop the server and client independently... so if you want to clear a cache or need to restart the frontend or backend independently you have to drop everything.

Did we want to note you can start them via `npm run start:dev` in the frontend and backend folders?

File: README.md [hunk 3]
```diff
@@ -8,83 +8,96 @@ A dashboard for Open Data Hub components.
...
+  - NPM recommended version -> `8.19.4`
 
 ### Additional tooling requirements
-* [OpenShift CLI, the "oc" command](https://docs.openshift.com/enterprise/3.2/cli_reference/get_started_cli.html#installing-the-cli)
-* [podman](https://github.com/containers/podman)
-* [Quay.io](https://quay.io/)
+
+- [OpenShift CLI, the "oc" command](https://docs.openshift.com/container-platform/4.12/cli_reference/openshift_cli/getting-started-cli.html)
+- [Podman](https://github.com/containers/podman)
+- [Quay.io](https://quay.io/)
```
Comment (andrewballantyne) [inline README.md:22]: Should we really be duplicating? Can the CONTRIB point at this?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Some thoughts -- all optional.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -3,74 +3,98 @@
...
-* [podman](https://github.com/containers/podman)
-* and have access to [Quay.io](https://quay.io/)
+
+- [NodeJS and NPM](https://nodejs.org/)
+  - Node recommended version -> `18.14.0`
+  - NPM recommended version -> `8.19.4`
+- [OpenShift CLI](https://docs.openshift.com/container-platform/4.12/cli_reference/openshift_cli/getting-started-cli.html)
+- [kustomize](https://github.com/kubernetes-sigs/kustomize)
+- [podman](https://github.com/containers/podman)
+- and have access to [Quay.io](https://quay.io/)
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply CONTRIBUTING.md:15]: Yes, that's a great idea, I'm gonna add it into another section though.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm!

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**